pkg/storage/replica.go

```diff
@@ -905,6 +905,10 @@ func (r *Replica) updateProposalQuotaRaftMuLocked(
 	// ourselves via looking at MsgAppResp messages would be overkill. Perhaps
 	// another accessor on RawNode.
 	status := r.raftStatusRLocked()
+	if status == nil {
```

Comment:
If the replica got destroyed, wouldn't the correct next step be destroying the quota pool as well? Otherwise (ok, and this is probably hard to get in practice) a replica could be destroyed but some goroutines are still waiting for quota and will never get it?